### PR TITLE
chore: Update ZSH completions (1.0.0+822a00c build)

### PR DESCRIPTION
**Pull request by Hanaasagi**
### What does this PR do?

Update bun's zsh completions. Part of #4016

#### Checklist

- [x] run
- [x] build
- [x] x
- [x] init
- [x] create
- [x] install
- [x] add
- [x] update
- [x] link
- [x] remove
- [x] unlink
- [x] pm
- [x] dev
- [x] upgrade
- [x] completions
- [x] discord
- [x] help

### How did you verify your code works?

Test manually.

### CODE REVIEW

**Comment by Hanaasagi** [comment]
Ready for review now. Each completion argument is based on the corresponding command's `--help`.

File: completions/bun.zsh [hunk 1]
```diff
@@ -1,350 +1,828 @@
...
+        '-u[Rewrite import URLs to start with --origin. Default: ""]:origin' \
+        '--port[Port to serve bun'"'"'s dev server on. Default: '"'"'3000'"'"']:port' \
+        '-p[Port to serve bun'"'"'s dev server on. Default: '"'"'3000'"'"']:port' \
+        '--smol[Use less memory, but run garbage collection more often]' \
+        '--minify[Minify (experimental)]' \
+        '--minify-syntax[Minify syntax and inline data (experimental)]' \
+        '--minify-identifiers[Minify identifiers]' \
+        '--no-macros[Disable macros from being executed in the bundler, transpiler and runtime]' \
+        '--target[The intended execution environment for the bundle. "browser", "bun" or "node"]: :->target' \
+        '--inspect[Activate Bun'"'"'s Debugger]:inspect' \
```
Comment (Hanaasagi) [inline completions/bun.zsh:604]: bun test --help displays so many parameters, but I think some of them are not being used.



Does the test runner need --inspect?

File: completions/bun.zsh [hunk 2]
```diff
@@ -354,300 +832,137 @@ _bun() {
...
     IFS=$'\n' bins=($(SHELL=zsh bun getcompletes b))
-    
+
     _alternative "scripts:scripts:(($scripts_list))"
     _alternative "bin:bin:(($bins))"
     _alternative "files:file:_files -g '*.(js|ts|jsx|tsx|wasm)'"
 }
 
+_bun_link_param_package_completion() {
+    # Read packages from ~/.bun/install/global/node_modules
```
Comment (Hanaasagi) [inline completions/bun.zsh:852]: I see that the package is stored in this directory after linked.

File: completions/bun.zsh [hunk 3]
```diff
@@ -1,350 +1,752 @@
...
+            'upgrade\:"Get the latest version of bun" '
+            'help\:"Show all supported flags and commands" '
+        )
+        main_commands=($main_commands)
+        _alternative "$scripts" "args:command:(($main_commands))" "files:files:(($files_list))"
+
+        ;;
+    args)
+        case $line[1] in
+        add)
```
Comment (matthewyu01) [inline completions/bun.zsh]: You can change it to `add|a)` to also autocomplete for the alias like how it is in the [bash completions](https://github.com/oven-sh/bun/blob/main/completions/bun.bash#L135).
Other ones are `create|c` `install|i` and `remove|rm`.

Comment (Hanaasagi) [reply completions/bun.zsh]: Fixed. thanks.

**Comment by daniel-melanson** [comment]
Resolves #2503 as well.

**Pull request merged**